### ISSUE TITLE
TOC has unexpected behavior when not top of note

### ISSUE THREAD

**Issue by toddfoster** [laurent22/joplin#1602]
### Operating system
- macOS
- iOS

### Application
- Desktop
- Mobile

When I leave a title for my note at the very top (which is conveniently copied into the empty title of a new note: thanks!), and throw in a ```[toc]``` on the next line, I end up with _three_ tables of contents. If I remove that title so that the ```[toc]``` is at the very top of the note, then I get the expected behavior (only a single table of contents).

![Screen Shot 2019-05-28 at 14 50 15](https://user-images.githubusercontent.com/985093/58504325-54200300-8158-11e9-946a-60706b80e123.png)

**Comment by CalebJohn** [comment]
If you leave a newline before the `[toc]` you also get the correct behaviour.
e.g.
```
Title

[toc]

Body...
```

This is probably a bug with the [toc plugin](https://github.com/nagaozen/markdown-it-toc-done-right) that joplin uses.

**Comment by toddfoster** [comment]
That makes a certain kind of sense. I've experienced at least one markdown parsers than required newlines between different block elements. Joplin seems pleasantly tolerant (other than with toc!).

I made a quick test with the [demo.html](https://github.com/nagaozen/markdown-it-toc-done-right/blob/master/demo.html) from [markdown-it-toc-done-right](https://github.com/nagaozen/markdown-it-toc-done-right) without reproducing the issue, but that's a pretty different use case. 

This may not be an issue worth pursuing: a note in a doc might be helpful, though. Or this (closed) bug report might be adequate.
